### ISSUE TITLE
Allow for multiple connections to Sonarr (and possibly Plex)

### ISSUE THREAD

**Issue by CollinHeist** [CollinHeist/TitleCardMaker#280]
The ability to connect to multiple instances of Sonarr would be nice for users that segregate their Sonarr instances for things like 4K/regular, or Anime/non-Anime shows.

This would obviously be a pretty big overhaul in both the YAML files as well as the code. My first thought for specifying via the preferences file would be something like:

```yaml
sonarr:
- url: # ...
  api_key: # ....
- url: # ...
  api_key: # ....
```

So a list of `SonarrInterface` objects could be created and utilized.

The next logistical hurdle is whether shows whose episode data is sources via Sonarr. I think the best option would be to query for series on all given interfaces until found, and don't allow or require the user to specify a specific server. The only downside I can imagine right now is that if a given series was present on both instances, but had different data - i.e. whether it was downloaded, any assigned tags, the media location - the _first_ server's data would always take priority. I guess this could be remedied by permitting a per-series `sonarr_server_id` YAML attribute that is the index of the server in the preferences file? For example:

```yaml
# preferences.yml
sonarr:
- url: 192.168.0.1:8989
  api_key: abcdef
- url: 192.168.0.2:8989
  api_key: 987654

# tv.yml
series:
  Mr. Robot (2015):
    # config info
    sonarr_server_id: 1
```

This would tell TCM to not bother querying the `192.168.0.1` server, as the `192.168.0.2` server is index/ID 1. 

---

Multiple Plex servers could be specified in preferences the same way, but that would certainly **require** specification of the Plex server ID due in part to how slow the Plex API is, and the fact that a Library already has to be specified per-series to link with Plex, so it could just be added there. Like so:

```yaml
# tv.yml
libraries:
  TV Shows:
    plex_server_id: 0
    path: ./tv/
  Anime:
    plex_server_id: 1
    path: ./anime/
```

**Comment by CollinHeist** [comment]
Merged in 3c17381913785bb3f482076d31638e3603395e71

**Comment by CollinHeist** [comment]
For now, I am not implementing/supporting multiple Plex servers. The complexity associated with that was too much, and so far I think the use would be very little
